mii/client.py

```diff
@@ -55,51 +74,189 @@ class MIIClient():
             modelresponse_pb2.google_dot_protobuf_dot_empty__pb2.Empty())
 
     def terminate(self):
         self.asyncio_loop.run_until_complete(self.terminate_async())
 
     async def create_session_async(self, session_id):
-        return await self.stub.CreateSession(
+        return await self.mr_stub.CreateSession(
             modelresponse_pb2.SessionID(session_id=session_id))
 
-    def create_session(self, session_id):
-        assert self.task == Tasks.TEXT_GENERATION, f"Session creation only available for task '{Tasks.TEXT_GENERATION}'."
+    def create_session(self, session_id, deployment_name=None):
+        if len(self.deployments > 1):
+            assert deployment_name is not None, "Deployment name must be passed in to create session when there are multiple models"
+        deployment_name, task = self._get_deployment_task(deployment_name)
+        assert task == Tasks.TEXT_GENERATION, f"Session creation only available for task '{Tasks.TEXT_GENERATION}'."
         return self.asyncio_loop.run_until_complete(
             self.create_session_async(session_id))
 
     async def destroy_session_async(self, session_id):
-        await self.stub.DestroySession(modelresponse_pb2.SessionID(session_id=session_id)
-                                       )
+        await self.mr_stub.DestroySession(
+            modelresponse_pb2.SessionID(session_id=session_id))
 
-    def destroy_session(self, session_id):
-        assert self.task == Tasks.TEXT_GENERATION, f"Session deletion only available for task '{Tasks.TEXT_GENERATION}'."
+    def destroy_session(self, session_id, deployment_name=None):
+        if len(self.deployments > 1):
+            assert deployment_name is not None, "Deployment name must be passed in to destroy session when there are multiple models"
+        deployment_name, task = self._get_deployment_task(deployment_name)
+        assert task == Tasks.TEXT_GENERATION, f"Session deletion only available for task '{Tasks.TEXT_GENERATION}'."
         self.asyncio_loop.run_until_complete(self.destroy_session_async(session_id))
 
 
+class LBClient(MIIClient):
+    def __init__(self,
+                 deployments,
+                 host,
+                 port,
+                 lb_config=None,
+                 model_path=None,
+                 port_map=None,
+                 deployment_tag=None):
+        super().__init__(deployments, host, port)
+        self.lb_stub = None
+        if port is not None:
+            channel = create_channel(host, port) if not self.channel else self.channel
+            self.lb_stub = modelresponse_pb2_grpc.DeploymentManagementStub(channel)
+        self.lb_config = lb_config
+        self.model_path = model_path
+        self.port_map = port_map if port_map is not None else {}
+        self.deployment_tag = deployment_tag
+
+    async def add_models_async(self, proto_request):
+        await getattr(self.lb_stub, "AddDeployment")(proto_request)
+
+    def add_models(self,
```

Comment:
I'm a bit confused about the implementation of this method. It looks like we are re-generating the score file when we add models. And then we call `init` again? Does this not cause problems for models that are already running?